sdk/identity/azure-identity/src/main/java/com/azure/identity/DeviceCodeCredential.java

```diff
@@ -34,30 +36,75 @@
      * @param identityClientOptions the options for configuring the identity client
      */
     DeviceCodeCredential(String clientId, String tenantId, Consumer<DeviceCodeInfo> challengeConsumer,
-                         IdentityClientOptions identityClientOptions) {
+                         boolean automaticAuthentication, IdentityClientOptions identityClientOptions) {
         this.challengeConsumer = challengeConsumer;
         identityClient = new IdentityClientBuilder()
             .tenantId(tenantId)
             .clientId(clientId)
             .identityClientOptions(identityClientOptions)
             .build();
         this.cachedToken = new AtomicReference<>();
+        this.authorityHost = identityClientOptions.getAuthorityHost();
+        this.automaticAuthentication = automaticAuthentication;
+        if (identityClientOptions.getAuthenticationRecord() != null) {
+            cachedToken.set(new MsalAuthenticationAccount(identityClientOptions.getAuthenticationRecord()));
+        }
     }
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
         return Mono.defer(() -> {
             if (cachedToken.get() != null) {
-                return identityClient.authenticateWithMsalAccount(request, cachedToken.get().getAccount())
+                return identityClient.authenticateWithMsalAccount(request, cachedToken.get())
                     .onErrorResume(t -> Mono.empty());
             } else {
                 return Mono.empty();
             }
         }).switchIfEmpty(
-            Mono.defer(() -> identityClient.authenticateWithDeviceCode(request, challengeConsumer)))
+            Mono.defer(() -> {
+                if (!automaticAuthentication) {
+                    return Mono.error(new AuthenticationRequiredException("Interactive authentication is needed to "
+                              + "acquire token. Call Authenticate to initiate the device code authentication."));
+                }
+                return identityClient.authenticateWithDeviceCode(request, challengeConsumer);
+            }))
             .map(msalToken -> {
-                cachedToken.set(msalToken);
+                cachedToken.set(
+                        new MsalAuthenticationAccount(
+                                new AuthenticationRecord(msalToken.getAuthenticationResult(),
+                                        identityClient.getTenantId())));
                 return msalToken;
             });
     }
+
+    /**
+     * Interactively authenticates a user via the default browser.
```

Comment:
Does `DeviceCodeCredential` open a browser?

Reply:
nope, user needs to open the browser.
The docs can be elaborated to explain that.
I'll do it.